pkg/controllers/subnetset/subnetset_controller.go

```diff
@@ -49,94 +49,100 @@ type SubnetSetReconciler struct {
 }
 
 func (r *SubnetSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	obj := &v1alpha1.SubnetSet{}
-	log.Info("reconciling subnetset CR", "subnetset", req.NamespacedName)
+	startTime := time.Now()
+	defer func() {
+		log.Info("Finished reconciling SubnetSet", "SubnetSet", req.NamespacedName, "time", time.Since(startTime))
+	}()
+
+	subnetsetCR := &v1alpha1.SubnetSet{}
 	metrics.CounterInc(r.SubnetService.NSXConfig, metrics.ControllerSyncTotal, MetricResTypeSubnetSet)
 
-	if err := r.Client.Get(ctx, req.NamespacedName, obj); err != nil {
-		log.Error(err, "unable to fetch subnetset CR", "req", req.NamespacedName)
-		return ResultNormal, client.IgnoreNotFound(err)
+	if err := r.Client.Get(ctx, req.NamespacedName, subnetsetCR); err != nil {
+		// IgnoreNotFound returns nil on NotFound errors.
+		if client.IgnoreNotFound(err) == nil {
+			if err := r.deleteSubnetBySubnetSetName(req.Name, req.Namespace); err != nil {
+				log.Error(err, "failed to delete NSX Subnet", "SubnetSet", req.NamespacedName)
+				return ResultRequeue, err
+			}
+			return ResultNormal, nil
+		}
+		log.Error(err, "unable to fetch Subnet CR", "req", req.NamespacedName)
+		return ResultRequeue, err
+	}
+	if err := r.cleanStaleSubnetsForSubnetSet(subnetsetCR.Name, subnetsetCR.Namespace, string(subnetsetCR.UID)); err != nil {
```

Comment:
It may have the same issue as cleanStaleSubnets in subnet controller?

Reply:
Yes, we don't need `cleanStaleSubnetsForSubnetSet` logic here, we should do it inside `client.IgnoreNotFound(err) == nil` condition.